Analytics branch

gaholt opened the pull request.

Commit:
added solution

Commit:
added to program.cs

Commit:
Added additional functions

Commit:
added functions for data ingress

Commit:
Completed ingress functions

Commit:
added correct json parsing

Commit:
fixed json parse bug

Commit:
Elimated bad using statements

Commit:
attempt to fix configuration problem

Commit:
removed null

Commit:
dispose configuration

Commit:
eliminated newtonsoft code

Commit:
changed variable names/ tightened up code

Commit:
updated readme

Commit:
made changes based on group review

Commit:
added to create sdstype function

Commit:
eliminated using statements, added async prefix to tasks, and improved other syntax

gaholt committed:
Code cleanup

Code cleanup that includes:

* Created Constants class for capturing all the magic strings and numbers used throughout the code

* Corrected code to conform with OSIsoft standards

* Fixed a bug related to not awaiting an async method

* Fixed a number of hidden resource leaks

* Fixed issue with not properly reusing HttpClient

* Removed appsettings.json since it was overkill

* Helper classes store timestamps as DateTime instead of string (that's not a pattern we want to promote).

* Updated readme.md to reflect removing the appsettings.json file.

dcbroad3 commented on EDSAnalytics/Coverity.proj:
```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="utf-8"?>
```
If you copy the [analysis template](https://github.com/osisoft/OSI-Samples-OMF/blob/master/miscellaneous/build_templates/analysis.yml) used by the other projects into this repo and reference that, it should run Coverity (as well as BlackDuck/BinSkim) and this file shouldn't be necessary. (we are in the process of migrating to Polaris anyway)

dcbroad3 commented on EDSAnalytics/EDSAnalytics.sln:
```diff
@@ -0,0 +1,31 @@
+﻿
```
Typically we add a `Solution Items` folder to the solution file with common sample files like the README, pipeline, etc, see [here](https://github.com/osisoft/OSI-Samples-OCS/blob/master/basic_samples/Authentication/AuthorizationCodeFlow/DotNet/AuthorizationCodeFlow.sln) for an example.

dcbroad3 commented on EDSAnalytics/EDSTest/EDSAnalytics.csproj:
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
We have been trying to add StyleCop and FxCop analyzers to our projects, see [this project](https://github.com/osisoft/OSI-Samples-OCS/tree/master/basic_samples/Authentication/ClientCredentialFlow/DotNet) for an example. If you don't have the bandwidth to do it, we can either work on the branch to do it or go back and add that later.

dcbroad3 commented on EDSAnalytics/EDSTest/README.md:
```diff
@@ -0,0 +1,105 @@
+# EDS Analytics Sample and Test
+
+[![Build Status](https://PLACE_BUILD_STATUS_LINK_HERE)]
```
We can add this if you'd like, but first I think the `azure-pipeline.yml` file needs to be moved up one level in the folder hierarchy.

dcbroad3 commented on EDSAnalytics/EDSTest/README.md:
```diff
@@ -0,0 +1,105 @@
+
+## Requirements
+
+- .NET Core 3.1 is installed
```
I'd combine the `## Requirements` and `### Prerequisites` sections

dcbroad3 commented on EDSAnalytics/EDSTest/README.md:
```diff
@@ -0,0 +1,105 @@
+# EDS Analytics Sample and Test
+
```
Add `**Version:**: 1.0.0` between header and build status badge

gaholt replied:
Done.

dcbroad3 commented on EDSAnalytics/EDSTest/README.md:
```diff
@@ -0,0 +1,105 @@
+Step 12 
+- The Types and Streams created by the application are deleted.
+
+
```
We usually add a common list of links to the bottom of this file, see [this README](https://github.com/osisoft/OSI-Samples-OCS/blob/master/basic_samples/Authentication/ClientCredentialFlow/DotNet/README.md) for an example

gaholt replied:
Done.

dcbroad3 commented on EDSAnalytics/EDSTest/README.md:
```diff
@@ -0,0 +1,105 @@
+# EDS Analytics Sample and Test
```
I think the `EDSTest` and `EDSTestTests` folders should be renamed to `EDSAnalytics` and `EDSAnalyticsTests` respectively

gaholt replied:
Fixed.

dcbroad3 commented on EDSAnalytics/EDSTest/README.md:
```diff
@@ -0,0 +1,105 @@
+# EDS Analytics Sample and Test
```
This README file should be moved up one level in the folder hierarchy.

dcbroad3 commented on EDSAnalytics/EDSTest/azure-pipeline.yml:
```diff
@@ -0,0 +1,72 @@
+trigger:
```
This file should be moved up one level in the folder hierarchy

dcbroad3 commented on EDSAnalytics/EDSTest/azure-pipeline.yml:
```diff
@@ -0,0 +1,72 @@
+        workingDirectory: $(Build.SourcesDirectory)$(testPath)/
+        displayName: 'Run tests'
+
+  - template: '/miscellaneous/build_templates/analysis.yml'
```
This references a file that has not yet been added to the repository. You can copy it (and the other files in the `/miscellaneous/build_templates/` folder) from [here](https://github.com/osisoft/OSI-Samples-OMF/blob/master/miscellaneous/build_templates/analysis.yml). Without all three files (config/variables/analysis) this build pipeline won't work.

gaholt replied:
Done.

dcbroad3 commented on EDSAnalytics/EDSTest/azure-pipeline.yml:
```diff
@@ -0,0 +1,72 @@
+          fileLibrary: $(CredFileLibrary)
+          filePath: $(projPath)
+
+      - script: |
```
There's actually a much cleaner way to run/test the app, see [this pipeline](https://github.com/osisoft/OSI-Samples-OCS/blob/master/basic_samples/Authentication/ClientCredentialFlow/DotNet/azure-pipeline.yml#L54) for an example

dcbroad3 commented on EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj:
```diff
@@ -0,0 +1,21 @@
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.2.0" />
+    <PackageReference Include="xunit" Version="2.4.0" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.0" />
+    <PackageReference Include="coverlet.collector" Version="1.0.1" />
```
Is `coverlet` actually used? Looks neat, not something I've used before.

gaholt replied:
Looks like that was copied forward from old unit tests.

gaholt replied:
Removed.

dcbroad3 commented on EDSAnalytics/EDSTestTests/EDSAnalyticsTest.csproj:
```diff
@@ -0,0 +1,21 @@
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.1.3" />
```
Reminder that if you do add StyleCop and FxCop analyzers, they need to be added to this project as well.

dcbroad3 commented on EDSAnalytics/EDSTest/README.md:
```diff
@@ -0,0 +1,105 @@
+# EDS Analytics Sample and Test
```
This sample needs a HISTORY.md file, see [HISTORY.md](https://github.com/osisoft/OSI-Samples-OCS/blob/master/basic_samples/Authentication/AuthorizationCodeFlow/DotNet/HISTORY.md) for an example. Right now should just include 1.0.0 / {date}, Initial public release

gaholt replied:
Done.

gaholt committed:
Create history.md

Added History.md

gaholt committed:
PR updates

Removed coverlet from test csproj
Added 3 missing files.

gaholt committed:
More PR cleanup

gaholt committed:
Correction

Moved History.md to the correct location
Moved Readme.md to the correct location.

gaholt committed:
Updates

Updated readme
Updated .gitattributes and .gitignore

Commit:
Merge remote-tracking branch 'origin/master' into AnalyticsBranch

Commit:
Reorganize folder structure

Commit:
Move azure-pipeline, formatting on README

Commit:
Readme rewording, status badge

Commit:
Code analysis and cleanup

Commit:
Add polaris.yml

Commit:
Finish code cleanup

Commit:
Clean up README files

Commit:
Minor cleanup

Commit:
Update version date

Commit:
Cleanup READMEs
